diesel/src/sqlite/connection/into_sqlite_result.rs

```diff
@@ -0,0 +1,141 @@
+use std::ffi::{CString, CStr};
+use std::os::raw as libc;
+
+use super::ffi;
+
+// TODO: Support BLOB values
+
+pub trait IntoSqliteResult {
```

Comment:
Do we need a separate trait for this? Can we use `ToSql` instead?

Reply:
I'm not sure. I feel that I don't know `ToSql` well enough to give a really decisive response. Again, the point here is to be able to use the correct sqlite API calls, in this case `sqlite3_result_*`. These functions are not called anywhere else in the code base.

Now I'm thinking that you might be mainly concerned about the additional `trait`, not the implementation. I started out trying to make it work with `FromSql` and `ToSql` and couldn't make it. I think they are the slightly wrong abstraction. However, it might be that somebody more familiar with the Diesel code base could be able to make it work.

Reply:
`ToSql` always just writes some bytes to a buffer. In the case of `Sqlite`, we then look at the `SqliteType` variant for each value, and figure out the right bind function to call based on that. We should be able to do pretty much the exact same thing here, but calling `sqlite3_result_*` instead.

Reply:
> However, it might be that somebody more familiar with the Diesel code base could be able to make it work.

Yeah, this is a pretty hairy feature to take on for your first PR